### ISSUE TITLE
gloveConsole blocks on close() if device still streams

### ISSUE THREAD

**Issue by guihomework** [ubi-agni/TactileGlove#5]
Similar problem as in `agni_serial_protocol` (which code derived from this package), if a device still streams while a `close()` on the serial port is called, this `close` is blocked for the duration of **close_wait**, unless another SIGTERM is called.  So always double SIGTERM needs to be issued (which is annoying)

The problem is known, see https://stackoverflow.com/questions/14255314/close-function-on-serial-port-descriptor-blocks-in-linux
or 
https://stackoverflow.com/questions/69261649/serial-port-file-descriptor-close-delay

The forum both suggest to use a system command to reduce the close wait time on the serial port.

I tested the `sudo setserial /dev/ttyACM0 closing_wait 100` which then made the app close almost immediately, but the suggested change of the `CRTSCTS` flags on the `oldtio` before closing did not produce the desired effect. I would have preferred an "internal" solution inside our code rather than a system setting.

Do you have another idea than an system setting to change that slow closing. 
It happens on devices that still stream (old glove protocol) or with the AGNI Serial Protocol in case a stop command failed to stop the stream.

**Comment by rhaschke** [comment]
> Similar problem as in `agni_serial_protocol`

Was this reported somewhere? If so, a link would be helpful.





> Do you have another idea than an system setting to change that slow closing.

Did you try the last answer [here](https://stackoverflow.com/questions/14255314/close-function-on-serial-port-descriptor-blocks-in-linux)?

**Comment by guihomework** [comment]
> > Similar problem as in `agni_serial_protocol`
> 
> Was this reported somewhere? If so, a link would be helpful.

no, I just discovered both today, while testing P4 gloves (using old protocol) with agni_serial_protocol (which should "gracefully" fail) and then with gloveConsole which should work. Both had the close blocked issue but have the same code for serial com.

> 
> > Do you have another idea than an system setting to change that slow closing.
> 
> Did you try the last answer [here](https://stackoverflow.com/questions/14255314/close-function-on-serial-port-descriptor-blocks-in-linux)?

if by last answer you mean the one with flowcontrol change, I applied `oldtio.c_cflag &= ~CRTSCTS;`  during init after `oldtio` is saved and this should be equivalent to the answer you point to because `oldtio` gets restored before closing the port.  I again, had anticipated your question but apparently was not clear in my initial message mentioning the `CRTSCTS`.

I suppose the flowcontrol changed help that user because he might have had a real serial port with real control lines, which if disabled would maybe stop the distant device from streaming data ?

**Comment by rhaschke** [comment]
I see. Do you mind issuing a `system()` call from source? Question: does one really require sudo privileges?
